packages/authorization-updater/src/utils.ts

```diff
@@ -103,3 +120,54 @@ export const getPurposeVersionFromEvent = (
 
   return { purposeId: purpose.id, purposeVersion };
 };
+
+export const getClientFromEvent = (
+  msg: {
+    data: {
+      client?: ClientV2;
+    };
+  },
+  eventType: string
+): Client => {
+  if (!msg.data.client) {
+    throw missingKafkaMessageDataError("client", eventType);
+  }
+
+  return fromClientV2(msg.data.client);
+};
+
+export const apiClientKind = {
+  consumer: "CONSUMER",
+  api: "API",
+} as const;
+export const ApiClientKind = z.enum([
+  Object.values(apiClientKind)[0],
+  ...Object.values(apiClientKind).slice(1),
+]);
+export type ApiClientKind = z.infer<typeof ApiClientKind>;
+
+export const clientKindToApiClientKind = (kid: ClientKind): ApiClientKind =>
+  match<ClientKind, ApiClientKind>(kid)
+    .with(clientKind.consumer, () => "CONSUMER")
+    .with(clientKind.api, () => "API")
+    .exhaustive();
+
+export const keyUseToApiKeyUse = (kid: KeyUse): ApiKeyUse =>
+  match<KeyUse, ApiKeyUse>(kid)
+    .with(keyUse.enc, () => "ENC")
+    .with(keyUse.sig, () => "SIG")
+    .exhaustive();
+
+export const descriptorStateToClientState = (
+  state: DescriptorState
+): ApiClientComponentState =>
+  state === descriptorState.published || state === descriptorState.deprecated
+    ? ApiClientComponent.Values.ACTIVE
+    : ApiClientComponent.Values.INACTIVE;
+
+export const purposeStateToClientState = (
+  state: PurposeVersionState
+): ApiClientComponentState =>
+  state === purposeVersionState.active
+    ? ApiClientComponent.Values.ACTIVE
+    : ApiClientComponent.Values.INACTIVE;
```

Comment:
There are parts in the `index.ts` file where the check is done without using these functions, could you update those parts using these?

Reply:
Updated